### PR TITLE
Week 3 - weekend challenge

### PR DESCRIPTION
**Pull request by UpstatePedro**

### CODE REVIEW

File: views/result.erb [hunk 1]
```diff
@@ -0,0 +1,15 @@
+<% if $game.player1.choice && $game.player2.choice %>
```
Comment (silvabox) [inline views/result.erb]: All of the logic in this view should be shared between the game and controller

File: views/game.erb [hunk 2]
```diff
@@ -0,0 +1,10 @@
+<h2><%= "#{session[:player_name]}, select your move!" %></h2>
```
Comment (silvabox) [inline views/game.erb]: The view should not know anything about the session

File: spec/game_spec.rb [hunk 3]
```diff
@@ -0,0 +1,35 @@
+require 'game'
+
+describe Game do
+  let(:player_nil) {double (:player), choice: nil}
+  let(:player_rock) {double (:player), choice: :Rock}
+  let(:player_scissors) {double (:player), choice: :Scissors}
+  let(:player_paper) {double (:player), choice: :Paper}
+
+  describe '#check_winner' do
+    it '#check_winner returns an error if either player has not made a choice' do
```
Comment (silvabox) [inline spec/game_spec.rb]: you already have a `describe '#check_winner'` block - you don't need to repeat `#check_winner` in your test description.

File: spec/game_spec.rb [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+      subject = described_class.new(player_nil, player_scissors)
+      expect{subject.check_winner}.to raise_error "Not all players have made their selection!"
+    end
+
+    it '#check_winner returns player1 (rock>scissors)' do
```
Comment (silvabox) [inline spec/game_spec.rb]: are you testing for player 2 winning?

File: spec/game_spec.rb [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+      subject = described_class.new(player_rock, player_scissors)
+      expect(subject.check_winner).to be(player_rock)
+    end
+
+    it '#check_winner returns player1 (scissors>paper)' do
+      subject = described_class.new(player_scissors, player_paper)
```
Comment (silvabox) [inline spec/game_spec.rb]: I would avoid using `subject` and `described_class` in this context - it's unwieldy - just use `game = Game.new(...)`

File: spec/player_spec.rb [hunk 6]
```diff
@@ -0,0 +1,18 @@
...
+  describe '#set_choice' do
+    it 'turns the string provided into a symbol with the correct value' do
+      selection =  "Paper"
+      subject.set_choice selection
+      expect(subject.choice).to be(:Paper)
+    end
+
+    it 'raises an error if the value provided is not valid' do
+      selection = "Badger"
+      expect(subject.set_choice selection).to eq('That is not a valid choice')
```
Comment (silvabox) [inline spec/player_spec.rb]: This is not an error being raised - it's just returning a string message.

File: views/index.erb [hunk 7]
```diff
@@ -0,0 +1,5 @@
+<h2>To begin playing, please enter your name:</h2>
```
Comment (silvabox) [inline views/index.erb]: Your `views` folder is better off in your `app` folder.

**Comment by silvabox** [comment]
Hi Piers, thanks for submitting your RPS challenge.  You've done a pretty good job - there are a few minor issues which I've commented on, but overall it's nice and clean and simple.  Please also refer to the [common issues](https://github.com/makersacademy/rps-challenge/wiki/Common-issues) as some of these apply to your code.

:checkered_flag:

**Comment by UpstatePedro** [comment]
Thanks Ben. Only discovered the existence of the "common issues" this week so I'll be using it more in future!!

**Pull request closed without merging**